### PR TITLE
Sk testing products

### PR DESCRIPTION
**Pull request by sienna-kopf**
# Crate Pull Request Form

## Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
- Testing for all product queries complete. Small bug fixed in terms of a Sequelize function ordering elements. Jest and Super test used, testing database created (CREATE DATABASE crate_test;) , migrated, and seeded (NODE_ENV=test npm run setup:db)  for use. To run `npm test`. 

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other, Please explain: Backfilling testing
`Your message here`
## Has This Been Tested?
- [ ] No
- [x] Yes
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration:
Install Jest `npm install --save-dev jest` </br>
Add to `package.json` 
```
"scripts": {
   "test": "jest --watch --runInBand --detectOpenHandles"
}
```
Install Supertest `npm install --save-dev supertest`</br>
Enter a `psql` session and create test database. (CREATE DATABASE crate_test;) </br>
Configure test database in database.json 
```
"test": {
    "username": "postgres",
    "password": null,
    "database": "crate_test",
    "host": "127.0.0.1",
    "dialect": "postgres",
    "seederStorage": "sequelize"
  }
```
Migrate and seed test database `NODE_ENV=test npm run setup:db` </br>
Run tests using `npm test`</br>

# Checklist:
- [X] My code has no unused/commented out code
- [X] My code has no console logs
- [X] I have performed a self-review of my own code
-  [X] I have commented on my code, particularly in hard-to-understand areas

### CODE REVIEW

File: code/api/src/modules/product/query.test.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      .post('/')
+      .send({ query: '{ productById(productId: 1) { id, name, description, slug, type, gender, image } }' })
+      .expect(200)
+    expect(response.body.data.productById.id).toEqual(1)
+    expect(response.body.data.productById.name).toEqual("Belt for Women")
+    expect(response.body.data.productById.description).toEqual("A very nice belt for women.")
+    expect(response.body.data.productById.slug).toEqual("belt-for-women",)
+    expect(response.body.data.productById.type).toEqual(2)
+    expect(response.body.data.productById.gender).toEqual(2)
+    expect(response.body.data.productById.image).toEqual("/images/stock/belt-female.jpg")
```
Comment (rwilliams659) [inline code/api/src/modules/product/query.test.js:41]: Nice job testing all the specific properties of productById object - super thorough

File: code/api/src/modules/product/resolvers.js [hunk 2]
```diff
@@ -38,7 +38,7 @@ export async function getRelated(parentValue, { productId }) {
       id: { [models.Sequelize.Op.not]: productId }
     },
     limit: 3,
-    order: [[models.Sequelize.fn('RAND')]] // mock related products by showing random products
+    order: [[models.Sequelize.fn('RANDOM')]] // mock related products by showing random products
```
Comment (rwilliams659) [inline code/api/src/modules/product/resolvers.js:41]: GREAT bug catch!

**Review by rwilliams659 — APPROVED** [review]
Even more great testing! Go team!

File: code/api/src/modules/product/resolvers.js (hunk 2 above)
Comment (Strompy) [reply code/api/src/modules/product/resolvers.js:41]: 🚫🐛

File: code/api/src/modules/product/query.test.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      .send({ query: '{ product(slug: "belt-for-women" ) { id, name, description, slug, type, gender, image } }' })
+      .expect(200)
+    expect(response.body.data.product.id).toEqual(1)
+    expect(response.body.data.product.name).toEqual("Belt for Women")
+    expect(response.body.data.product.description).toEqual("A very nice belt for women.")
+    expect(response.body.data.product.slug).toEqual("belt-for-women",)
+    expect(response.body.data.product.type).toEqual(2)
+    expect(response.body.data.product.gender).toEqual(2)
+    expect(response.body.data.product.image).toEqual("/images/stock/belt-female.jpg")
+  })
```
Comment (Strompy) [inline code/api/src/modules/product/query.test.js:56]: Thorough test!

**Review by Strompy — APPROVED** [review]

File: code/api/src/modules/product/query.test.js (hunk 3 above)
Comment (JoshSevy) [reply code/api/src/modules/product/query.test.js:56]: Great work! I will definitely be using this for a reference in the future when testing.

**Review by JoshSevy — APPROVED** [review]
Great work! Love this team.

File: code/api/src/modules/product/query.test.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+  it("returns a product by Id", async () => {
+    const response = await request(server)
+      .post('/')
+      .send({ query: '{ productById(productId: 1) { id, name, description, slug, type, gender, image } }' })
+      .expect(200)
+    expect(response.body.data.productById.id).toEqual(1)
+    expect(response.body.data.productById.name).toEqual("Belt for Women")
+    expect(response.body.data.productById.description).toEqual("A very nice belt for women.")
+    expect(response.body.data.productById.slug).toEqual("belt-for-women",)
```
Comment (Abdeboskey) [inline code/api/src/modules/product/query.test.js:38]: Ok, now it makes sense what that slug is doing on the front end! I think!

File: code/api/src/modules/product/query.test.js (hunk 3 above)
Comment (Abdeboskey) [reply code/api/src/modules/product/query.test.js:56]: Agreed, thank you for putting this work in!

File: code/api/src/modules/product/query.test.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  })
+
+  it("returns related products", async () => {
+    const response = await request(server)
+      .post('/')
+      .send({ query: '{ productsRelated(productId: 1) { id, name, description, slug, type, gender, image } }' })
+      .expect(200)
+    expect(response.body.data.productsRelated.length).toEqual(3)
+  })
+
```
Comment (Abdeboskey) [inline code/api/src/modules/product/query.test.js:65]: How are the products related?

**Review by Abdeboskey — APPROVED** [review]
Really impressed by y'all's JS prowess. These test suites look 👌🏼👌🏼👌🏼

**Review by mlynch5187 — APPROVED** [review]

File: code/api/src/modules/product/query.test.js (hunk 5 above)
Comment (sienna-kopf) [reply code/api/src/modules/product/query.test.js:65]: Randomly!

**Pull request merged**